ts-rs/src/lib.rs

```diff
@@ -630,11 +633,14 @@ macro_rules! impl_tuples {
             fn inline() -> String {
                 panic!("tuple cannot be inlined!");
             }
-            fn dependency_types() -> impl TypeList
+            fn visit_generics(v: &mut impl TypeVisitor)
             where
                 Self: 'static
             {
-                ()$(.push::<$i>())*
+                $(
+                    v.visit::<$i>();
+                    <$i>::visit_generics(v);
+                )*
```

Comment:
That should fix the tuple issue. There's no `visit_dependencies` right now - As far as I can tell, that'd only be relevant if `inline` did anything - for example inlining all contents of the tuple, like we do for `Vec`.